dataextractor.py

```diff
@@ -0,0 +1,300 @@
+            elif stereo == oechem.OEBondStereo_Trans:
+                print('trans')
+                stereo_bonds.append((rd_n1, rd_a1, rd_a2, rd_n2, False))
+
+                # add bond stereochemistry:
+    for (rda1, rda2, rda3, rda4, is_cis) in stereo_bonds:
+        # get neighbor bonds
+        bond1 = rdmol.GetBondBetweenAtoms(rda1, rda2)
+        bond2 = rdmol.GetBondBetweenAtoms(rda3, rda4)
+
+        # Since this is relative, the first bond always goes up
+        # as explained above these names come from SMILES slashes so UP/UP is Trans and Up/Down is cis
+        bond1.SetBondDir(Chem.BondDir.ENDUPRIGHT)
+        if is_cis:
+            bond2.SetBondDir(Chem.BondDir.ENDDOWNRIGHT)
+        else:
+            bond2.SetBondDir(Chem.BondDir.ENDUPRIGHT)
+
+    # if oemol has coordinates (The dimension is non-zero)
+    # add those coordinates to the rdmol
+    if oechem.OEGetDimensionFromCoords(oemol) > 0:
+        conformer = Chem.Conformer()
+        oecoords = oemol.GetCoords()
+        for oe_idx, rd_idx in map_atoms.items():
+            (x,y,z) = oecoords[oe_idx]
+            conformer.SetAtomPosition(rd_idx, Geometry.Point3D(x,y,z))
+        rdmol.AddConformer(conformer)
+
+    # Save the molecule title
+    rdmol.SetProp("_Name", oemol.GetTitle())
+
+    # Cleanup the rdmol
+    # Note I copied UpdatePropertyCache and GetSSSR from Shuzhe's code to convert oemol to rdmol here:
+    rdmol.UpdatePropertyCache(strict=False)
+    Chem.GetSSSR(rdmol)
+    # I added AssignStereochemistry which takes the directions of the bond set
+    # and assigns the stereochemistry tags on the double bonds
+    Chem.AssignStereochemistry(rdmol, force=False)
+
+    print("Final Molecule: ", Chem.MolToSmiles(Chem.RemoveHs(rdmol), isomericSmiles=True))
+    return rdmol.GetMol()
+
+
+def make_molname_df(directory,ffdirectorylist):
+    """This is the make_molname_df function. It finds the names of mol2 files
+    that are present in all forcefield-minimized directories in the directory.
+    Its output is a pandas dataframe. 
+    """
+    dname = {}
+    # for every forcefield, create dataframe of molnames
+    for ffdirect in ffdirectorylist:
+        tempdirectory = '%s/%s/' % (directory, ffdirect)
+        list_of_mol2 = sorted(os.listdir(tempdirectory))
+        list_of_molnames = list()
+        for mol2 in list_of_mol2:
+            molName = mol2.split('.')[0]
+            list_of_molnames.append(molName)
+        tempnamedf = pd.DataFrame(np.array(list_of_molnames))
+        dname["%s" % ffdirect] = tempnamedf
+    # merges all dataframes generated above into one dataframe
+    # only keeps molecules present in all dataframes
+    twonamedf = dname[ffdirectorylist[0]].merge(dname[ffdirectorylist[1]])
+    for i in range(2, len(ffdirectorylist)):
+        threenamedf = dname[ffdirectorylist[i]].merge(twonamedf)
+    threenamedf = threenamedf.rename({0: "MolNames"},axis='columns')
+    return threenamedf
+
```

Comment:
Should check that at least 2 forcefields are present in the ffdirect

Reply:
Add an Issue for this?